fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java

```diff
@@ -235,4 +238,64 @@ protected void runAfterCatalogReady() {
         deleteUnusedShardAndShardGroup();
         deleteUnusedWorker();
     }
+
+    // do a one time sync, delete all shards from this table that exist in starmgr but not in fe
+    public static void syncTableMeta(String dbName, String tableName) throws DdlException {
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
+        if (db == null) {
+            throw new DdlException(String.format("db %s does not exist.", dbName));
+        }
+
+        HashMap<Long, List<Long>> feGroupToShards = new HashMap<>();
+        db.readLock();
+        try {
+            Table table = db.getTable(tableName);
+            if (table == null) {
+                throw new DdlException(String.format("table %s does not exist.", tableName));
+            }
+            if (!table.isCloudNativeTableOrMaterializedView()) {
+                throw new DdlException("only support cloud table or cloud mv.");
+            }
+
+            OlapTable olapTable = (OlapTable) table;
+            for (Partition partition : olapTable.getAllPartitions()) {
+                for (PhysicalPartition physicalPartition : partition.getSubPartitions()) {
+                    long groupId = physicalPartition.getShardGroupId();
+                    List<Long> feShardIds = new ArrayList<>();
+                    for (MaterializedIndex materializedIndex :
+                            physicalPartition.getMaterializedIndices(MaterializedIndex.IndexExtState.ALL)) {
+                        for (Tablet tablet : materializedIndex.getTablets()) {
+                            feShardIds.add(tablet.getId());
+                        }
+                    }
+                    if (!feShardIds.isEmpty()) {
+                        feGroupToShards.put(groupId, feShardIds);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new DdlException(e.getMessage());
+        } finally {
+            db.readUnlock();
+        }
+
+        // delete tablet meta and data, outside db lock
+        StarOSAgent starOSAgent = GlobalStateMgr.getCurrentStarOSAgent();
+        Set<Long> shardToDelete = new HashSet<>();
+        for (Map.Entry<Long, List<Long>> entry : feGroupToShards.entrySet()) {
+            List<Long> starmgrShardIds = starOSAgent.listShard(entry.getKey());
+            starmgrShardIds.removeAll(entry.getValue());
+            try {
+                // delete shard in starmgr but not in fe
+                dropTabletAndDeleteShard(starmgrShardIds, starOSAgent);
+            } catch (Exception e) {
+                // ignore exception
+                LOG.info(e.getMessage());
+            }
+            shardToDelete.addAll(starmgrShardIds);
```

Comment:
consider delete the shards from starmgr right after the dropTabletAndDeleteShard() success, or depends on the size of shardToDelete, to avoid the bad case that the for loop takes long long time, and the final delete may delay for long time without progress.

Reply:
done

Reply:
add force parameter